/* Copyright 2022 The MediaPipe Authors. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto2";

package mediapipe.tasks.components.processors.proto;

option java_package = "com.google.mediapipe.tasks.components.processors.proto";
option java_outer_classname = "DetectorOptionsProto";

// Shared options used by all detection tasks.
message DetectorOptions {
  // The locale to use for display names specified through the TFLite Model
  // Metadata, if any. Defaults to English.
  optional string display_names_locale = 1 [default = "en"];

  // The maximum number of top-scored detection results to return. If < 0,
  // all available results will be returned. If 0, an invalid argument error is
  // returned.
  optional int32 max_results = 2 [default = -1];

  // Score threshold, overrides the ones provided in the model metadata
  // (if any). Results below this value are rejected.
  optional float score_threshold = 3;

  // Overlapping threshold for non-maximum-suppression calculator. Only used for
  // models without built-in non-maximum-suppression, i.e., models that don't
  // use the Detection_Postprocess TFLite Op
  optional float min_suppression_threshold = 6;

  // Optional allowlist of category names. If non-empty, detections whose
  // category name is not in this set will be filtered out. Duplicate or unknown
  // category names are ignored. Mutually exclusive with category_denylist.
  repeated string category_allowlist = 4;

  // Optional denylist of category names. If non-empty, detection whose category
  // name is in this set will be filtered out. Duplicate or unknown category
  // names are ignored. Mutually exclusive with category_allowlist.
  repeated string category_denylist = 5;

  // Whether to use multiclass NMS. That is, each category processes
  // non-maximum-suppression separately.
  optional bool multiclass_nms = 7 [default = false];
}
